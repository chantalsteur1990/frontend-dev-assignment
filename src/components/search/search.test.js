import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Search from './search';

const mockProps = {
  name: 'search',
  placeholder: 'Zoeken',
  ariaLabelInput: 'zoeken',
  ariaLabelClear: 'annuleren',
  ariaLabelSubmit: 'zoeken',
  handleChange: jest.fn(),
  handleClear: jest.fn(),
  handleSubmit: jest.fn(),
  showButtonClear: true
};

it('should render (snapshot)', () => {
  const component = shallow(<Search {...mockProps} />);
  expect(shallowToJson(component)).toMatchSnapshot();
});
