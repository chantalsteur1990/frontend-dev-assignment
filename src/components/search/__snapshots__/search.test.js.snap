// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render (snapshot) 1`] = `
<div
  className="input-search"
>
  <div
    className="input-search__actions"
  >
    <button
      aria-label="annuleren"
      onClick={[Function]}
    >
      <img
        alt="icon-clear"
        className="input-search__actions__icon"
        src="clear.svg"
      />
    </button>
    <button
      aria-label="zoeken"
      onClick={[Function]}
    >
      <img
        alt="icon-search"
        className="input-search__actions__icon"
        src="search.svg"
      />
    </button>
  </div>
  <input
    aria-label="zoeken"
    className="input-search__input"
    name="search"
    onChange={[Function]}
    placeholder="Zoeken"
    type="text"
  />
</div>
`;
